pinot-core/src/main/java/org/apache/pinot/core/data/manager/realtime/RealtimeSegmentDataManager.java

```diff
@@ -828,20 +828,26 @@ public void run() {
               _segmentLogger.info("Could not commit segment. Retrying after hold");
               hold();
               break;
+            }
             default:
               _segmentLogger.error("Holding after response from Controller: {}", response.toJsonString());
               hold();
               break;
           }
         }
       } catch (Exception e) {
-        String errorMessage = "Exception while in work";
-        _segmentLogger.error(errorMessage, e);
-        postStopConsumedMsg(e.getClass().getName());
-        _state = State.ERROR;
-        _realtimeTableDataManager.addSegmentError(_segmentNameStr, new SegmentErrorInfo(now(), errorMessage, e));
-        _serverMetrics.setValueOfTableGauge(_clientId, ServerGauge.LLC_PARTITION_CONSUMING, 0);
-        return;
+        if (_shouldStop) {
+          _segmentLogger.info("Caught exception in consumer thread after stop() is invoked: {}, ignoring the exception",
+              e.toString());
```

Comment:
This will not print the stack trace. Shouldn't be better to print it?

Reply:
I intentionally log it as info and hide the stack trace because this is normal scenario